src/user_database.ts

```diff
@@ -188,15 +196,23 @@ export class PrismaUserDatabase implements UserDatabase {
     return result;
   }
 
-  async query<R>(sql: string, ...params: any[]): Promise<R[]> {
+  async queryFunction<C extends UserDatabaseClient, R, T extends unknown[]>(func: UserDatabaseQuery<C, R, T>, ...args: T): Promise<R> {
```

Comment:
I think for Prisma, we can simply pass in a `PrismaClient` obeject. For example, we may simply pass in the PrismaClient and it will not be transactional:

```js
return await func(this.prisma, ...args);
```

However, I'm not sure how we can disallow updates.

Reply:
Thanks.